src/mesh/RTree.cpp

```diff
@@ -5,6 +5,8 @@ namespace mesh {
 
 // Initialize static member
 std::map<int, rtree::PtrRTree> precice::mesh::rtree::trees;
+// Initialize static member
+std::map<int, PtrPrimitiveRTree> precice::mesh::rtree::_primitive_trees;
```

Comment:
Why store pointer to trees and not the tree itself, like how it's done with VertexTree?

Reply:
The cache of the `VertexRTree` holds `shared_ptr` and I decided to stay consistent with the current implementation.
```cpp
  using PtrRTree = std::shared_ptr<VertexRTree>;
// ...
  static std::map<int, PtrRTree> trees;
```

Reply:
My bad, you're right.